x/params/subspace/subspace.go

```diff
@@ -0,0 +1,198 @@
+package subspace
+
+import (
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Additional capicity to be allocated for Subspace.name
+// So we don't have to allocate extra space each time appending to the key
+const extraKeyCap = 20
```

Comment:
Thanks for doing this, its a great idea :)
I'd actually recommend making it `tmhash.Size` (the size of at least an address hash)